spec/services/flood_risk_engine/finalize_enrollment_spec.rb

```diff
@@ -0,0 +1,39 @@
+require "rails_helper"
+require "byebug"
+
+module FloodRiskEngine
+  RSpec.describe FinalizeEnrollmentService, type: :service do
+    subject { described_class.new(enrollment) }
+    let(:enrollment) do
+      build_stubbed(:enrollment, status: Enrollment.statuses[:building])
+    end
+
+    context "with a nil enrollment" do
+      let(:enrollment) { nil }
+      it "raises an error" do
+        expect { subject.finalize! }.to raise_error(ArgumentError)
+      end
+    end
+
+    context "with an enrollment with a status other than 'building', "\
+            "for example one that already has a status of 'pending'" do
+      let(:enrollment) do
+        build_stubbed(:enrollment, status: Enrollment.statuses[:pending])
+      end
+      it "raises an error" do
+        expect { subject.finalize! }
+          .to raise_error(InvalidEnrollmentStateError)
+      end
+    end
+
+    it "changes the enrollment status to pending and sends the submitted email" do
```

Comment:
As this is the only test that uses the default `enrollment`, I'd move it to the top near to where enrollment is defined.